### PR TITLE
3: QA Review

### PR DESCRIPTION
**Pull request by lieut-data**
#### Summary
Update the code review guidelines to document that QA review now occurs after Dev review completes, as per the discussion at https://community-release.mattermost.com/core/pl/mjnexpqgetna5bfkm1xd9nhwjy.

Note that I have not yet effected the GitHub label change from `2: QA Review` -> `3: QA Review`, but will hopefully accomplish that shortly. There are also other automation tickets to be filed from the agreed changes.

Apologies for the noisy diff -- I took the opportunity to tidy things up given the scope of the necessary changes.

### CODE REVIEW

File: site/content/contribute/getting-started/code-review.md [hunk 1]
```diff
@@ -38,47 +38,65 @@ If you are a core committer seeking a review
...
+        - Don't be afraid to pick someone who gives "hard" reviews. Code review feedback is never a personal attack: it should "sharpen" the skills of both the author and the reviewers, not to mention improving the quality of the product.
+        - Try to avoid assigning the same person to all of your reviews unless they are related.
+        - When in doubt, ask for recommendations on our community server.
+    * [Wait](#if-you-are-awaiting-a-review) for their review to complete before continuing so as to avoid churn if changes are requested.
+    * Remove the `2: Dev Review` label only when these reviews are done.
+        - The last developer to review may choose to help here and speed up the process.
```
Comment (catalintomai) [inline site/content/contribute/getting-started/code-review.md]: maybe add more detail, like how would they be able to speed up/help?

File: site/content/contribute/getting-started/code-review.md [hunk 2]
```diff
@@ -99,11 +117,10 @@ If you are a core committer asked to give a review
...
     * Feel free to challenge assumptions and timelines. Rushing a change into a patch release may cause more harm than good.
 4. Merge the pull request.
-    * Do not merge until the reviewing product manager has approved the changes and removed the `1: PM Review` label.
-    * Do not merge until the reviewing QA tester has approved the changes and added the `QA Review Done` label.
-    * Do not merge until at least two core committers have approved and all concerns have been addressed.
-    * Remove any remaining `2: Dev Review` label and assign the `3: Reviews Complete` label.
+    * Do not merge until the `1: PM Review`, `2: Dev Review` and `3: QA Review` labels have been removed.
+    * Do not merge if there are outstanding changes requested.
+    * Do not merge if there are any `Do Not Merge` labels applied.
+        - When in doubt, leave the merging of the pull request to the author.
```
Comment (catalintomai) [inline site/content/contribute/getting-started/code-review.md]: Since this document is from the perspective of the author ("If you are a community member seeking a review"), maybe rephrase?

File: site/content/contribute/getting-started/code-review.md [hunk 3]
```diff
@@ -38,47 +38,65 @@ If you are a core committer seeking a review
...
+        - If in doubt, assign any QA tester and comment about the uncertainty. The QA tester may reassign as appropriate.
+    * [Wait](#if-you-are-awaiting-a-review) for their review to complete before continuing.
+        - It is the QA tester's responsibility to determine the scope of required testing, if any.
+    * Remove the `3: QA Review` label only when their review is done.
+        - This is typically the responsibility of the QA tester.
+        - The QA tester may add the `QA Review Done` label when their review is done, or if a review is deemed not necessary.
+    * In the rare event your changes do not require QA review, you may remove the `3: QA Review` label yourself.
+        - Comment on the pull request clearly explaining the rationale.
+8. Merge the pull request.
+    * Do not merge until the `1: PM Review`, `2: Dev Review` and `3: QA Review` labels have been removed.
```
Comment (catalintomai) [inline site/content/contribute/getting-started/code-review.md]: "labels have been removed" - can they be removed by somebody else than the author? If no, maybe we can rephrase this to smth. like "you have removed", to be more explicit.

File: site/content/contribute/getting-started/code-review.md [hunk 4]
```diff
@@ -38,47 +38,65 @@ If you are a core committer seeking a review
...
+    * Remove the `2: Dev Review` label only when these reviews are done.
+        - The last developer to review may choose to help here and speed up the process.
+7. After Dev review, assign a [QA tester](/contribute/getting-started/core-committers/#qa-testers).
+    * The choice of QA tester is up to you.
+        - In most cases, choose the QA tester embedded with your team.
+        - If your change primarily touches another team's codebase, consider their QA tester.
+        - If in doubt, assign any QA tester and comment about the uncertainty. The QA tester may reassign as appropriate.
+    * [Wait](#if-you-are-awaiting-a-review) for their review to complete before continuing.
+        - It is the QA tester's responsibility to determine the scope of required testing, if any.
+    * Remove the `3: QA Review` label only when their review is done.
```
Comment (catalintomai) [inline site/content/contribute/getting-started/code-review.md]: "Remove the" - maybe rephrase to smth like : "The QA tester removes the label..." (with the exception below).

File: site/content/contribute/getting-started/code-review.md [hunk 5]
```diff
@@ -38,47 +38,65 @@ If you are a core committer seeking a review
...
     * Most issues are targeted for an upcoming release, and should be assigned the corresponding milestone.
     * The milestone is mandatory for bug fixes that must be cherry-picked.
-7. Wait for a review.
+5. Request a review from a [product manager](/contribute/getting-started/core-committers/#product-managers).
+    * The choice of product manager is up to you.
+        - In most cases, choose the product mananger embedded with your team.
+        - If your change primarily touches another team's codebase, consider their product mananger.
+        - If in doubt, assign any product mananger and comment about the uncertainty. The product mananger may reassign as appropriate.
+    * [Wait](#if-you-are-awaiting-a-review) for their review to complete before continuing so as to avoid churn if changes are requested.
+    * Remove the `1: PM Review` label only when their review is done.
```
Comment (catalintomai) [inline site/content/contribute/getting-started/code-review.md]: How do they communicate that? QA has the 'QA Review Done' label.

File: site/content/contribute/getting-started/code-review.md [hunk 6]
```diff
@@ -38,47 +38,65 @@ If you are a core committer seeking a review
...
+        - Product managers ensure the changes meet [user experience guidelines](https://docs.mattermost.com/developer/fx-guidelines.html).
+    * If your changes do not affect the user experience, you may remove `1: PM Review` immediately.
```
Comment (hanzei) [inline site/content/contribute/getting-started/code-review.md]: I have the cut feel that 95% of all PR's are merged without PM Review. And this doesn't seam to be an issue. What is the reasoning on adding `1: PM Review` by default, if it gets removed immediately most of the cases?

File: site/content/contribute/getting-started/code-review.md [hunk 7]
```diff
@@ -38,47 +38,65 @@ If you are a core committer seeking a review
...
+7. After Dev review, assign a [QA tester](/contribute/getting-started/core-committers/#qa-testers).
+    * The choice of QA tester is up to you.
+        - In most cases, choose the QA tester embedded with your team.
+        - If your change primarily touches another team's codebase, consider their QA tester.
+        - If in doubt, assign any QA tester and comment about the uncertainty. The QA tester may reassign as appropriate.
+    * [Wait](#if-you-are-awaiting-a-review) for their review to complete before continuing.
+        - It is the QA tester's responsibility to determine the scope of required testing, if any.
+    * Remove the `3: QA Review` label only when their review is done.
+        - This is typically the responsibility of the QA tester.
+        - The QA tester may add the `QA Review Done` label when their review is done, or if a review is deemed not necessary.
```
Comment (hanzei) [inline site/content/contribute/getting-started/code-review.md]: Will QA still use `QA Review Done`?

Comment (lieut-data) [reply site/content/contribute/getting-started/code-review.md]: My understanding was that this was still under discussion. @lindalumitchell, do you have any new context into the plans for `QA Review Done`? Perhaps we just omit the mention from this document altogether, since it's not applicable to the reviewer.

File: site/content/contribute/getting-started/code-review.md (hunk 6 above)
Comment (lieut-data) [reply site/content/contribute/getting-started/code-review.md]: I have the same gut feel. @wiersgallak, we've long since pretended that `1:PM Review` preceded `2:Dev Review`, but that rarely /seems/ to happen. With these changes, it's obviously a little more painful to pretend.

Thoughts on the future of `1:PM Review`? Should it start as a blocking item as proposed, with it being removed in many cases right away? Does this help improve visibility into the need for PM Review on certain issues (as we hope these changes continue to do for QA Review)?

File: site/content/contribute/getting-started/code-review.md (hunk 5 above)
Comment (lieut-data) [reply site/content/contribute/getting-started/code-review.md]: Good question -- I had been assuming a "GitHub review" gets completed in this case, but it would be useful to call that out explicitly.

File: site/content/contribute/getting-started/code-review.md (hunk 4 above)
Comment (lieut-data) [reply site/content/contribute/getting-started/code-review.md]: Yeah, I waffled on this point a bit too much. I feel like removing the `3: QA Review` label shouldn't treated differently than any other -- if QA has approved, removing the label can be done by anyone.

@lindalumitchell, thoughts?

File: site/content/contribute/getting-started/code-review.md (hunk 2 above)
Comment (lieut-data) [reply site/content/contribute/getting-started/code-review.md]: This phrasing in actually under the `If you are a core committer asked to give a review` section -- do you think the language `author` makes sense here?

File: site/content/contribute/getting-started/code-review.md (hunk 1 above)
Comment (lieut-data) [reply site/content/contribute/getting-started/code-review.md]: Agreed! Or, perhaps I can just move that advice to the section on core community members giving a review.

File: site/content/contribute/getting-started/code-review.md (hunk 2 above)
Comment (catalintomai) [reply site/content/contribute/getting-started/code-review.md]: My bad, I missed the header. In this context, "author" makes sense to me. - "community member" (seeking a review) would be prob. too verbose(?).

File: site/content/contribute/getting-started/code-review.md [hunk 8]
```diff
@@ -38,47 +38,65 @@ If you are a core committer seeking a review
...
-    * If you are Mattermost staff, try to take into account the timeoff calendar.
-    * Try to avoid assigning the same person to all of your reviews unless they are related.
-    * When in doubt, ask for recommendations on our community server.
-4. Assign a [QA tester](/contribute/getting-started/core-committers/#qa-testers) and label your pull request with `2: QA Review`.
-    * It is the PM or QA tester's responsibility to determine the scope of required testing, if any.
-    * Reviews by QA may occur at the same time as review by core committers. Be sure to ask for a second review as needed when changes are made.
-    * The QA tester should remove the `2: QA Review` label and add the `QA Review Done` label when their review is done, or if a review is deemed not necessary.
-5. Apply additional labels as necessary:
+2. Immediately assign `1: PM Review`, `2: Dev Review`, and `3: QA Review`.
+    * Your pull request should not be merged until these labels are later removed.
```
Comment (wiersgallak) [inline site/content/contribute/getting-started/code-review.md]: Recommend a clarification to this: Your pull request should not be merged until each type of reviewer has reviewed and approved the PR and all review labels are removed.

File: site/content/contribute/getting-started/code-review.md (hunk 4 above)
Comment (catalintomai) [reply site/content/contribute/getting-started/code-review.md]: The comment was made 'cause we say "This is typically the responsibility of the QA tester." and further down "In the rare event..you (the author) may remove", I was just thinking that we should advise for the most common case.

File: site/content/contribute/getting-started/code-review.md (hunk 6 above)
Comment (wiersgallak) [reply site/content/contribute/getting-started/code-review.md]: I think it depends on the change.  When there is a new feature and or a change that changes the user experience (text change, new feature) than the PM should be reviewing from the customers perspective  - does this meet the requirements, do what it was intended to do, make the purpose/instruction clear and easy to follow.  

I do like visibility on the changes even if I don't have to explicitly approve them, especially when the change is focused on the backend, the Dev Review is likely to proceed without input from me required. In backend changes, I do not want to be a blocker before Dev reviews get started, because I actually get value from seeing the developers' comments.

File: site/content/contribute/getting-started/code-review.md (hunk 8 above)
Comment (lieut-data) [reply site/content/contribute/getting-started/code-review.md:42]: I'm worried about ambiguity here, such as when `1: PM Review` or `3: QA Review` is skipped as per the process outlined in later points, and no such reviewer is (intentionally) included. Is the concern here that someone might just remove the labels and merge?

File: site/content/contribute/getting-started/code-review.md (hunk 4 above)
Comment (lindalumitchell) [reply site/content/contribute/getting-started/code-review.md:77]: Yes, I agree that the most common case would make sense to state here, especially with exception noted below. Could be something like "The QA tester removes the `3: QA Review` label when their review is done." @catalintomai @lieut-data

File: site/content/contribute/getting-started/code-review.md (hunk 7 above)
Comment (lindalumitchell) [reply site/content/contribute/getting-started/code-review.md:79]: Thanks, yes, we are still finalizing decisions about the labels we will use for visibility, alerting, and reporting on the QA side. I think it is fine to remove the line about `QA Review Done` entirely from this doc, and we can include it in QA-specific docs as needed.  @hanzei @lieut-data

File: site/content/contribute/getting-started/code-review.md [hunk 9]
```diff
@@ -38,47 +38,65 @@ If you are a core committer seeking a review
...
+    * [Wait](#if-you-are-awaiting-a-review) for their review to complete before continuing so as to avoid churn if changes are requested.
+    * Remove the `2: Dev Review` label only when these reviews are done.
+        - The last developer to review may choose to help here and speed up the process.
+7. After Dev review, assign a [QA tester](/contribute/getting-started/core-committers/#qa-testers).
+    * The choice of QA tester is up to you.
+        - In most cases, choose the QA tester embedded with your team.
+        - If your change primarily touches another team's codebase, consider their QA tester.
+        - If in doubt, assign any QA tester and comment about the uncertainty. The QA tester may reassign as appropriate.
+    * [Wait](#if-you-are-awaiting-a-review) for their review to complete before continuing.
+        - It is the QA tester's responsibility to determine the scope of required testing, if any.
```
Comment (lindalumitchell) [inline site/content/contribute/getting-started/code-review.md:76]: Would it make sense to include a note that in some cases QA review might happen in parallel with PM and/or Dev reviews? Considered suggesting it earlier in the doc, like in the PM section, but given that it is an exception, and a QA-focused exception, maybe it would fit in this section. 

Something like "In some cases it may be agreed during planning that QA should review in parallel with PM or Dev reviews." ?

File: site/content/contribute/getting-started/code-review.md [hunk 10]
```diff
@@ -38,47 +38,65 @@ If you are a core committer seeking a review
...
-    * Don't be afraid to pick someone who gives "hard" reviews. Code review feedback is never a personal attack: it should "sharpen" the skills of both the author and the reviewers, not to mention improving the quality of the product.
-    * If you are Mattermost staff, try to take into account the timeoff calendar.
-    * Try to avoid assigning the same person to all of your reviews unless they are related.
-    * When in doubt, ask for recommendations on our community server.
-4. Assign a [QA tester](/contribute/getting-started/core-committers/#qa-testers) and label your pull request with `2: QA Review`.
-    * It is the PM or QA tester's responsibility to determine the scope of required testing, if any.
-    * Reviews by QA may occur at the same time as review by core committers. Be sure to ask for a second review as needed when changes are made.
-    * The QA tester should remove the `2: QA Review` label and add the `QA Review Done` label when their review is done, or if a review is deemed not necessary.
-5. Apply additional labels as necessary:
+2. Immediately assign `1: PM Review`, `2: Dev Review`, and `3: QA Review`.
```
Comment (lindalumitchell) [inline site/content/contribute/getting-started/code-review.md:41]: Perhaps "Immediately add labels" rather than "Immediately assign", in order to make explicitly distinct from assigning reviewers?

File: site/content/contribute/getting-started/code-review.md (hunk 7 above)
Comment (lindalumitchell) [reply site/content/contribute/getting-started/code-review.md:79]: Oh, what about the `4: Reviews Complete` label; still using that one, right? Useful to include a mention here, maybe something like "The QA tester adds the label `4: Reviews Complete` when all reviews are done." to replace the line about `QA Review Done`?

File: site/content/contribute/getting-started/code-review.md (hunk 3 above)
Comment (lindalumitchell) [inline site/content/contribute/getting-started/code-review.md:83]:
```suggestion
    * Do not merge until the labels `1: PM Review`, `2: Dev Review` and `3: QA Review` have been removed and the label `4: Reviews Complete` has been added.
```

File: site/content/contribute/getting-started/code-review.md (hunk 2 above)
Comment (lieut-data) [reply site/content/contribute/getting-started/code-review.md:123]: I /think/ author suffices here, since it might be a community member or another core committer.

File: site/content/contribute/getting-started/code-review.md (hunk 6 above)
Comment (lieut-data) [reply site/content/contribute/getting-started/code-review.md:59]: Let's leave the `1: PM Review` unchanged for now.

File: site/content/contribute/getting-started/code-review.md (hunk 7 above)
Comment (lieut-data) [reply site/content/contribute/getting-started/code-review.md:79]: I admit I'm not keen on the `4: Reviews Complete` label. We haven't used it consistently in the past, and it effectively duplicates information already present in the pull request between the requested reviews and blocking labels.

Thoughts on just omitting it altogether?

File: site/content/contribute/getting-started/code-review.md (hunk 9 above)
Comment (lieut-data) [reply site/content/contribute/getting-started/code-review.md:76]: I think it makes sense to stick to the happy path for documentation purposes -- already so much text here.

**Comment by lieut-data** [comment]
Thanks for the reviews, @lindalumitchell & @hanzei! Updated and comments above -- let me know your thoughts on the outstanding items.

File: site/content/contribute/getting-started/code-review.md (hunk 8 above)
Comment (wiersgallak) [reply site/content/contribute/getting-started/code-review.md:42]: Ah - I see that this is further clarified below under each type of review.  To me, this statement was a bit obsure.  Maybe even just clarifying, "Your pull requests should not be merged until these labels are later removed in the review process."

Comment (lieut-data) [reply site/content/contribute/getting-started/code-review.md:42]:
```suggestion
    * Your pull request should not be merged until these labels are later removed in the review process.
```

File: site/content/contribute/getting-started/code-review.md (hunk 7 above)
Comment (hanzei) [reply site/content/contribute/getting-started/code-review.md]: To me `4: Reviews Complete` serves as a "green light" from QA that all necessary reviews had happend. But I see that it's cumbersome work and it's not used consistently.

I'm fine with dropping it. Maybe we don't remove the label (jet) so we can return to the label if needed.

File: site/content/contribute/getting-started/labels.md [hunk 11]
```diff
@@ -27,8 +27,8 @@ We leverage [GitHub labels](https://help.github.com/en/articles/about-labels) to
 * `1: PM Review`: Requires review by a [product manager](/contribute/getting-started/core-committers/#product-managers).
```
Comment (catalintomai) [inline site/content/contribute/getting-started/labels.md:27]: nit: do we need the indices 1,2,3,4 for the PM/UX/.. Review labels (since we don't have them for the other labels in the set) and do we need to duplicate the index for PM and UX.

Comment (lieut-data) [reply site/content/contribute/getting-started/labels.md:27]: It's actually a sort of hack: this forces them to sort to the top of the list in the GitHub labels screen.

**Comment by lindalumitchell** [comment]
Thanks very much, @lieut-data! Also appreciate the tidy consolidation. All looks good to me, with the one outstanding item about `4: Reviews Complete` (my comment re: wanting to keep it in use is above, in-thread).

File: site/content/contribute/getting-started/code-review.md [hunk 12]
```diff
@@ -98,12 +113,14 @@ If you are a core committer asked to give a review
...
-4. Merge the pull request.
-    * Do not merge until the reviewing product manager has approved the changes and removed the `1: PM Review` label.
-    * Do not merge until the reviewing QA tester has approved the changes and added the `QA Review Done` label.
-    * Do not merge until at least two core committers have approved and all concerns have been addressed.
-    * Remove any remaining `2: Dev Review` label and assign the `3: Reviews Complete` label.
+4. Avoid leaving a review hanging.
+    * Try to accept or reject the review instead of just leaving comments.
+    * If you are the last developer to approve the changes, consider requesting a review from the appropriate QA tester to speed up the process.
+5. Merge the pull request.
+    * Do not merge until the `1: PM Review`, `2: Dev Review` and `3: QA Review` labels have been removed.
```
Comment (justinegeffen) [inline site/content/contribute/getting-started/code-review.md:120]:
```suggestion
    * Do not merge until the labels `1: PM Review`, `2: Dev Review` and `3: QA Review` labels have been removed and the label `4: Reviews Complete` has been added.
```

File: site/content/contribute/getting-started/labels.md [hunk 13]
```diff
@@ -27,8 +27,8 @@ We leverage [GitHub labels](https://help.github.com/en/articles/about-labels) to
...
 * `1: UX Review`: Requires review by a UX Designer.
```
Comment (justinegeffen) [inline site/content/contribute/getting-started/labels.md:28]:
```suggestion
* `1: UX Review`: Requires review by a UX designer.
```

File: site/content/contribute/getting-started/code-review.md [hunk 14]
```diff
@@ -38,47 +38,62 @@ If you are a core committer seeking a review
...
     * `Do Not Merge/Awaiting PR`: Apply this if the pull request depends on another (e.g. server changes)
     * `Setup Test Server`: Apply this to create a test server with your changes for review.
-    * See [labels](/contribute/getting-started/labels) for more details.
-6. Assign a milestone as necessary.
+    * See [labels](/contribute/getting-started/labels) for additional labels.
+4. Assign a milestone as necessary.
     * Most issues are targeted for an upcoming release, and should be assigned the corresponding milestone.
     * The milestone is mandatory for bug fixes that must be cherry-picked.
-7. Wait for a review.
+5. Request a review from a [product manager](/contribute/getting-started/core-committers/#product-managers).
```
Comment (justinegeffen) [inline site/content/contribute/getting-started/code-review.md:51]:
```suggestion
5. Request a review from a [Product Manager](/contribute/getting-started/core-committers/#product-managers).
```

File: site/content/contribute/getting-started/code-review.md [hunk 15]
```diff
@@ -38,47 +38,62 @@ If you are a core committer seeking a review
...
+    * The choice of product manager is up to you.
```
Comment (justinegeffen) [inline site/content/contribute/getting-started/code-review.md:52]:
```suggestion
    * The choice of Product Manager is up to you.
```

File: site/content/contribute/getting-started/code-review.md [hunk 16]
```diff
@@ -38,47 +38,62 @@ If you are a core committer seeking a review
...
+        - In most cases, choose the product mananger embedded with your team.
```
Comment (justinegeffen) [inline site/content/contribute/getting-started/code-review.md:53]:
```suggestion
        - In most cases, choose the Product Manager embedded with your team.
```

File: site/content/contribute/getting-started/code-review.md [hunk 17]
```diff
@@ -38,47 +38,62 @@ If you are a core committer seeking a review
...
+        - If your change primarily touches another team's codebase, consider their product mananger.
```
Comment (justinegeffen) [inline site/content/contribute/getting-started/code-review.md:54]:
```suggestion
        - If your change primarily touches another team's codebase, consider their Product Manager.
```

File: site/content/contribute/getting-started/code-review.md [hunk 18]
```diff
@@ -38,47 +38,62 @@ If you are a core committer seeking a review
...
+        - If in doubt, assign any product mananger and comment about the uncertainty. The product mananger may reassign as appropriate.
```
Comment (justinegeffen) [inline site/content/contribute/getting-started/code-review.md:55]:
```suggestion
        - If in doubt, assign any Product Manager and comment about the uncertainty. The Product Manager may reassign as appropriate.
```

File: site/content/contribute/getting-started/code-review.md [hunk 19]
```diff
@@ -38,47 +38,62 @@ If you are a core committer seeking a review
...
+    * [Wait](#if-you-are-awaiting-a-review) for their review to complete before continuing so as to avoid churn if changes are requested.
+    * Remove the `1: PM Review` label only when their review is done and they accept the changes.
+        - Product managers ensure the changes meet [user experience guidelines](https://docs.mattermost.com/developer/fx-guidelines.html).
```
Comment (justinegeffen) [inline site/content/contribute/getting-started/code-review.md:58]:
```suggestion
        - Product Managers ensure the changes meet [user experience guidelines](https://docs.mattermost.com/developer/fx-guidelines.html).
```

**Comment by justinegeffen** [comment]
I hope you don't mind me jumping in on this and adding a few minor suggestions, @lieut-data (I saw the editor review label when I was scanning through PRs) :). This looks great. :)

File: site/content/contribute/getting-started/code-review.md (hunk 7 above)
Comment (lieut-data) [reply site/content/contribute/getting-started/code-review.md]: Sounds good! I'll put `4: Reviews Complete` back into the document.

**Comment by lieut-data** [comment]
Thanks, @justinegeffen! @lindalumitchell, I've added a mention of `4: Reviews Finished`, though I didn't tie it directly to QA since it can be added by other reviewers or even the author as needed.

Reading through this, I'd love to see us add a section entitled, `If you are a QA tester asked to give a review` to codify the flow for both sides. Perhaps a future contribution?

**Comment by lindalumitchell** [comment]
@saturninoabril I removed the QA Review label before I realized that you're also assigned to review this PR. If you'd like to review it please go for it; if not, I think we're good to go from the QA perspective.

**Comment by saturninoabril** [comment]
Thanks, Linda. LGTM.
